agent/api/testutils/container_equal.go

```diff
@@ -82,12 +82,12 @@ func ContainersEqual(lhs, rhs *api.Container) bool {
 	if lhs.AppliedStatus != rhs.AppliedStatus {
 		return false
 	}
-	if lhs.KnownExitCode == nil || rhs.KnownExitCode == nil {
-		if lhs.KnownExitCode != nil || rhs.KnownExitCode != nil {
+	if lhs.GetKnownExitCode() == nil || rhs.GetKnownExitCode() == nil {
```

Comment:
Since this is in test code, it might not be that big of a deal. But, an assumption (however incorrect because of race-conditions) made here was that `lhs.KnownExitCode` and `rhs.KnownExitCode` do not change within the scope of this block. Can you do this instead?

```
lhsExitCode := lhs.GetKnownExitCode() 
rhsExitCode := rhs.GetKnownExitCode()

if lhsExitCode == nil || rhsExitCode == nil {
 if lhsExitCode != nil || rhsExitCode != nil {
 ..
 } else {
  if !lhsExitCode != *rhsExitCode {
   ..
  }
 }
...